sdk/src/main/java/io/opentelemetry/sdk/trace/export/SpanData.java

```diff
@@ -0,0 +1,410 @@
+   * @since 0.1.0
+   */
+  public abstract Resource getResource();
+
+  /**
+   * Returns the name of this {@code Span}.
+   *
+   * @return the name of this {@code Span}.
+   * @since 0.1.0
+   */
+  public abstract String getName();
+
+  /**
+   * Returns the kind of this {@code Span}.
+   *
+   * @return the kind of this {@code Span}.
+   * @since 0.1.0
+   */
+  public abstract Kind getKind();
+
+  /**
+   * Returns the start {@code Timestamp} of this {@code Span}.
+   *
+   * @return the start {@code Timestamp} of this {@code Span}.
+   * @since 0.1.0
+   */
+  public abstract Timestamp getStartTimestamp();
+
+  /**
+   * Returns the attributes recorded for this {@code Span}.
+   *
+   * @return the attributes recorded for this {@code Span}.
+   * @since 0.1.0
+   */
+  public abstract Map<String, AttributeValue> getAttributes();
+
+  /**
+   * Returns the timed events recorded for this {@code Span}.
+   *
+   * @return the timed events recorded for this {@code Span}.
+   * @since 0.1.0
+   */
+  public abstract List<TimedEvent> getTimedEvents();
+
+  /**
+   * Returns links recorded for this {@code Span}.
+   *
+   * @return links recorded for this {@code Span}.
+   * @since 0.1.0
+   */
+  public abstract List<io.opentelemetry.trace.Link> getLinks();
+
+  /**
+   * Returns the {@code Status}.
+   *
+   * @return the {@code Status}.
+   * @since 0.1.0
+   */
+  public abstract Status getStatus();
+
+  /**
+   * Returns the end {@code Timestamp}.
+   *
+   * @return the end {@code Timestamp}.
+   * @since 0.1.0
+   */
+  public abstract Timestamp getEndTimestamp();
+
+  /**
+   * A timed event representation.
+   *
+   * @since 0.1.0
+   */
+  @Immutable
+  @AutoValue
+  public abstract static class TimedEvent {
+    /**
+     * Returns a new immutable {@code TimedEvent<T>}.
+     *
+     * @param timestamp the {@code Timestamp} of this event.
+     * @param event the event.
+     * @return a new immutable {@code TimedEvent<T>}
+     * @since 0.1.0
+     */
+    public static TimedEvent create(Timestamp timestamp, io.opentelemetry.trace.Event event) {
+      return new AutoValue_SpanData_TimedEvent(timestamp, event.getName(), event.getAttributes());
+    }
+
+    /**
+     * Returns the {@code Timestamp} of this event.
+     *
+     * @return the {@code Timestamp} of this event.
+     * @since 0.1.0
+     */
+    public abstract Timestamp getTimestamp();
+
+    /**
+     * Returns the name of this event.
+     *
+     * @return the name of this event.
+     */
+    public abstract String getName();
+
+    /**
+     * Gets the attributes for this event.
+     *
+     * @return the attributes for this event.
+     */
+    public abstract Map<String, AttributeValue> getAttributes();
+
+    TimedEvent() {}
+  }
+
+  /**
+   * Creates a new Builder for creating an SpanData instance.
+   *
+   * @return a new Builder.
+   * @since 0.1.0
+   */
+  public static Builder newBuilder() {
+    return new AutoValue_SpanData.Builder();
```

Comment:
We should initialize the default values for non-nullable members, with empty list/maps or default values.

Reply:
yep, agreed. fix incoming